### PR TITLE
Special game Mixed ore map

### PR DESCRIPTION
**Pull request by developer-8**
Adds a new special game that covers the entire map in mixed ore. 3 patterns available.
Mixed ore
Screenshot from 2023-01-19 21-02-09
Checkerboard. The size of cells can be specified.
Screenshot from 2023-01-19 21-03-11
Vertical lines. By bits-orio
Screenshot from 2023-01-19 21-04-24

Vote https://discord.com/channels/823696400797138974/836401275645198357/1064920925528674304

### Tested Changes:
- [x] I've tested the changes locally and you should too.
- [ ] I've not tested the changes.

### CODE REVIEW

**Comment by developer-8** [comment]
Made several tests to compare the performance of chunk generation. 
Command line params example `factorio --benchmark save_file --benchmark-ticks N`
Surface settings for all tests are the same, without random values. Performance for the other mixed ore maps types is almost the same, so all tests are only for `mixed ore size 5`
| benchmark-ticks\type | default                                                                                  | mixed ore size 5                                                                         | checkboard size 5                                                                      | vertical lines                                                                         |
|------------|------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------|
|        100 | Performed 100 updates in 2335.879 ms   avg: 23.359 ms, min: 0.083 ms, max: 1589.891 ms   | Performed 100 updates in 5148.284 ms   avg: 51.483 ms, min: 0.079 ms, max: 4033.564 ms   |                                                                                        |                                                                                        |
|        300 |   Performed 300 updates in 3328.516 ms   avg: 11.095 ms, min: 0.077 ms, max: 1377.571 ms |   Performed 300 updates in 7965.906 ms   avg: 26.553 ms, min: 0.094 ms, max: 3965.647 ms | Performed 300 updates in 7865.164 ms   avg: 26.217 ms, min: 0.088 ms, max: 3788.564 ms | Performed 300 updates in 7990.177 ms   avg: 26.634 ms, min: 0.074 ms, max: 3991.796 ms |
|        500 | Performed 500 updates in 3777.655 ms   avg: 7.555 ms, min: 0.069 ms, max: 1532.455 ms    | Performed 500 updates in 9232.786 ms   avg: 18.466 ms, min: 0.071 ms, max: 4011.046 ms   | Performed 500 updates in 9317.152 ms   avg: 18.634 ms, min: 0.061 ms, max: 3799.751 ms | Performed 500 updates in 8993.009 ms   avg: 17.986 ms, min: 0.071 ms, max: 3848.034 ms |
|       1000 | Performed 1000 updates in 3665.129 ms   avg: 3.665 ms, min: 0.058 ms, max: 1335.286 ms   | Performed 1000 updates in 9483.632 ms   avg: 9.484 ms, min: 0.066 ms, max: 4260.109 ms   |                                                                                        |                                                                                        |
|      10000 | Performed 10000 updates in 4280.739 ms   avg: 0.428 ms, min: 0.038 ms, max: 1301.676 ms  | Performed 10000 updates in 9570.083 ms   avg: 0.957 ms, min: 0.036 ms, max: 3888.880 ms  |                                                                                        |              

I also tested this map by simply launching the game and running to the side to force the game to generate chunks. On my pc in a normal game, the UPS sank to values of 58, and with infinite ore settings to 55.
From this, I can conclude that the generation of chunks with infinite ore negatively affects performance, but at the same time, I hope the players will not move far from the center of the map, because with a abundance of ore there is no need to look for a good place for an outpost

File: maps/biter_battles_v2/game_over.lua [hunk 1]
```diff
@@ -335,6 +334,8 @@ function Public.server_restart()
                 end
                 Gui.create_main_gui(player)
             end
+            -- Special_games.generate_special_games()
```
Comment (Ragnarok77-factorio) [inline maps/biter_battles_v2/game_over.lua:337]: why was it added ? Mistake I guess and should be deleted..?

Comment (developer-8) [reply maps/biter_battles_v2/game_over.lua:337]: yes

**Comment by developer-8** [comment]
There is a bug where the mixed ore map does not generate the starting turrets and walls. I will fix later.

**Comment by Ragnarok77-factorio** [comment]
ok @developer-8
Waiting for your update.

File: comfy_panel/special_games.lua [hunk 2]
```diff
@@ -409,6 +429,29 @@ function Public.has_life(player_name)
...
+		if type == 1 and size > 10 then
+			size = 10
+		end
+		global.next_special_games["mixed_ore_map"] = true
+		global.next_special_games_variables["mixed_ore_map"] = {
+			type = type,
+			size = size
+		}
+
+		game.print("Special game Mixed ore map is being scheluded. The special game will start after restarting the map!", Color.warning)
```
Comment (Ragnarok77-factorio) [inline comfy_panel/special_games.lua:451]: Little typo for scheluded.

**Review by Ragnarok77-factorio — CHANGES_REQUESTED** [review]
Very minor thing, otherwise OK for me. so we should be

**Review by Ragnarok77-factorio — APPROVED** [review]
OK now

**Review by Ragnarok77-factorio — APPROVED** [review]
OK

**Comment by Ragnarok77-factorio** [comment]
No big issue reported, will be merged this week-end

**Pull request merged**